### PR TITLE
[IMP] pos_self_order: renaming field and bug fix

### PR DESCRIPTION
**Pull request by davidmonnom**
### First commit:

Here is the list of bugs that have been fixed:
- Search, when you click on the search button, the input isn't autofocused.

- The tag list now scrolls automatically so that the selected tag is always visible on the screen.

- The back button did not work when the self was opened from a page containing the back button. This is because we are using the browser history. In this case, when the user clicks back, we send them to the home page.

- Automatic tag selection during scrolling has been improved.

---
### Second commit:

Previously in the front-end there was a notion of "tag" which used product categories. This led to misunderstandings, because in the backend there was a notion of tag and category for the products.

Now, the name of the category field is aligned in the backend and the frontend.

The product model ID field has also been changed from product_id to id during loading.

---
### Third commit:
Improvement of several aspects of the self-order UX:
- Changed the sentence indicating that the self-order is menu only.
- Added a shadow to the header.
- Reduced the size of the product image in the product list.
- Reduced the size of the product title in the product list.
- Reduced the size of the product image when you are on the page to add it to the shopping basket.
- Added a shadow above the sticky area on a product's main page.
- The quantity of a product already added to the basket has been moved from next to the title to next to the price.
- The review button now floats above the product list and no longer has a background colour.
- In the basket, taxes and the total are now displayed after orderlines and no longer sticky at the bottom.
- When the quantity of a product in the cart is changed to 0, the button is now called "remove from cart".

taskId: 3323949

---
### Fourth commit:
Added protection for the order button so that it cannot be inadvertently clicked twice.

---
### Fifth commit:
Correction of the order in which categories are displayed in the self following the deletion of the "order" argument on the product search in a previous PR.

### CODE REVIEW

File: addons/pos_self_order/static/src/self_order_service.js [hunk 1]
```diff
@@ -67,7 +66,14 @@ export class SelfOrder {
...
 
-        this.productsGroupedByTag = groupBy(this.products, "tag");
+        this.productsGroupedByCategory = this.products.reduce((acc, product) => {
+            if (!acc[product.pos_categ_id.name]) {
+                acc[product.pos_categ_id.name] = [];
+            }
+
+            acc[product.pos_categ_id.name].push(product);
+            return acc;
+        }, {});
```
Comment (caburj) [inline addons/pos_self_order/static/src/self_order_service.js:76]:
```suggestion
        this.productsGroupedByCategory = groupBy(this.products, (product) => product.pos_categ_id.name)
```
I think you can keep the use of `groupBy`.

File: addons/pos_self_order/static/src/pages/order_cart/order_cart.js [hunk 2]
```diff
@@ -24,9 +25,15 @@ export class OrderCart extends Component {
...
+    async processOrder() {
+        if (this.sendInProgress) {
+            return;
+        }
+
         if (this.selfOrder.self_order_mode === "meal") {
-            this.selfOrder.sendDraftOrderToServer();
+            this.sendInProgress = true;
+            await this.selfOrder.sendDraftOrderToServer();
+            this.sendInProgress = false;
```
Comment (caburj) [inline addons/pos_self_order/static/src/pages/order_cart/order_cart.js:36]:
```suggestion
            try {
                await this.selfOrder.sendDraftOrderToServer();
            } finally {
                this.sendInProgress = false;
            }
```

Need to make sure it's set to `false` also when `sendDraftOrderToServer` throws an error.

Comment (davidmonnom) [reply addons/pos_self_order/static/src/pages/order_cart/order_cart.js:36]: good catch ! thx

**Comment by caburj** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**